easybuild/easyblocks/p/psmpi.py

```diff
@@ -0,0 +1,96 @@
+##
+# Copyright 2016-2016 Ghent University, Forschungszentrum Juelich
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing the ParaStationMPI library, implemented as an easyblock
+
+@author: Damian Alvarez (Forschungszentrum Juelich)
+"""
+import os
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.mpich import EB_MPICH
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_psmpi(EB_MPICH):
+    """
+    Support for building the ParaStationMPI library.
+    - some compiler dependent configure options
+    """
+
+    def configure_step(self):
+        """
+        Custom configuration procedure for ParaStationMPI
+        * unset environment variables that leak into mpi* wrappers, and define $MPICHLIB_* equivalents instead
+        """
+
+        # things might go wrong if a previous install dir is present, so let's get rid of it
+        if not self.cfg['keeppreviousinstall']:
+            self.log.info("Making sure any old installation is removed before we start the build...")
+            super(EB_psmpi, self).make_dir(self.installdir, True, dontcreateinstalldir=True)
+
+        # MPICH configure script complains when F90 or F90FLAGS are set,
+        # they should be replaced with FC/FCFLAGS instead.
+        # Additionally, there are a set of variables (FCFLAGS among them) that should not be set at configure time,
+        # or they will leak in the mpix wrappers.
+        # Specific variables to be included in the wrapper exists, but they changed between MPICH 3.1.4 and MPICH 3.2
+        # and in a typical scenario we probably don't want them.
+        self.setup_env_vars()
+
+        # Bypass the configure_step of EB_MPICH
+        super(EB_MPICH, self).configure_step()
```

Comment:
another option is to move the MPICH-specific configure options block into a method, and then redefining that method here as:

```python
def define_mpich_config_opts(self):
    """Skip updating configure options for MPICH derivates; doesn't apply to psmpi."""
    pass
```

and then you can leave the `configure_step` of MPICH untouched; this avoids copy-pasting the `keeppreviousinstall` but, and the need to skip calling out to the `configure_step` of your parent like you're doing here

I think this may be cleaner...